### PR TITLE
Sns pref for android

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/27392
Resolves https://github.com/brave/brave-browser/issues/27378

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
 - Added brave://flags/#brave-wallet-sns for Desktop and Android
 - Added SNS pref(hidden if feature flag is not set) for Android(similar to ENS and UD)
image

### CODE REVIEW

File: android/brave_java_sources.gni [hunk 1]
```diff
@@ -183,8 +183,8 @@ brave_java_sources = [
   "../../brave/android/java/org/chromium/chrome/browser/custom_layout/popup_window_tooltip/PopupWindowTooltip.java",
   "../../brave/android/java/org/chromium/chrome/browser/custom_layout/popup_window_tooltip/PopupWindowTooltipUtils.java",
   "../../brave/android/java/org/chromium/chrome/browser/decentralized_dns/settings/ENSSettingsFragment.java",
-  "../../brave/android/java/org/chromium/chrome/browser/decentralized_dns/settings/RadioButtonGroupENSResolveMethodPreference.java",
-  "../../brave/android/java/org/chromium/chrome/browser/decentralized_dns/settings/RadioButtonGroupUnstoppableDomainsResolveMethodPreference.java",
+  "../../brave/android/java/org/chromium/chrome/browser/decentralized_dns/settings/RadioButtonGroupDDnsResolveMethodPreference.java",
```
Comment (supermassive) [inline android/brave_java_sources.gni:186]: RadioButtonGroupDDnsResolveMethodPreference options are same for ENS, SNS and UD

**Review by AlexeyBarabash — APPROVED** [review]
++

File: android/java/org/chromium/chrome/browser/preferences/BravePrefServiceBridge.java [hunk 2]
```diff
@@ -220,6 +220,14 @@ public int getENSResolveMethod() {
         return BravePrefServiceBridgeJni.get().getENSResolveMethod();
     }
 
+    public void setSnsResolveMethod(int method) {
+        BravePrefServiceBridgeJni.get().setSnsResolveMethod(method);
+    }
+
+    public int getSnsResolveMethod() {
+        return BravePrefServiceBridgeJni.get().getSnsResolveMethod();
+    }
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/preferences/BravePrefServiceBridge.java:229]: Please avoid using `BravePrefServiceBridge` at all. We are going to retire it soon. We should use upstream's `UserPrefs` instead. Check that as an example https://source.chromium.org/chromium/chromium/src/+/main:chrome/android/java/src/org/chromium/chrome/browser/sync/settings/AccountManagementFragment.java;l=289?q=AccountManagementFragment.java&ss=chromium%2Fchromium%2Fsrc.
That is a place on where you add you cpp header enum https://github.com/brave/brave-core/blob/master/build/android/config.gni#L56 and that's where you populate your preference https://github.com/brave/brave-core/blob/master/browser/android/preferences/BUILD.gn#L35. After all of that you can access your pref from java similar to this `UserPrefs.get(Profile.getLastUsedRegularProfile())                                        .getBoolean(BravePref.USE_REWARDS_STAGING_SERVER)`

**Review by SergeyZhukovsky — APPROVED** [review]
++

File: android/java/org/chromium/chrome/browser/preferences/BravePrefServiceBridge.java (hunk 2 above)
Comment (supermassive) [reply android/java/org/chromium/chrome/browser/preferences/BravePrefServiceBridge.java:229]: Well I just copypasted from similar methods above :)

File: android/brave_java_resources.gni [hunk 3]
```diff
@@ -1,6 +1,7 @@
-# Copyright 2020 The Brave Authors. All rights reserved.
-# Use of this source code is governed by a BSD-style license that can be
-# found in the LICENSE file.
+# Copyright (c) 2022 The Brave Authors. All rights reserved.
```
Comment (goodov) [inline android/brave_java_resources.gni:1]: pls keep the existing copyright year.

**Review by goodov — APPROVED** [review]

**Pull request merged**